### PR TITLE
Trade: Use mypy annotation for cythonization instead of via .pyx file

### PR DESCRIPTION
**Pull request by rht**

### CODE REVIEW

**Comment by rht** [comment]
@DavoudTaghawiNejad, @DavoudTaghawiNejad, @DavoudTaghawiNejad the 4th commit (even though the change is minimum) speeds up the unittest runtime a lot:
```
python start.py  106.95s user 23.05s system 200% cpu 1:04.73 total
python start.py  107.39s user 23.18s system 201% cpu 1:04.83 total
```

**Comment by rht** [comment]
@DavoudTaghawiNejad this is the optimal level of annotation. Precisifying further (e.g. replacing sender's `object` with `tuple`) etc only makes it worse.

File: abcEconomics/trade.py [hunk 1]
```diff
@@ -40,19 +40,23 @@
...
+epsilon: cython.double = 0.00000000001
 
 
 def get_epsilon():
     return epsilon
 
 
+# @cython.cclass
 class Offer(object):
-    __slots__ = ('sender', 'receiver', 'good', 'quantity', 'price', 'currency',
```
Comment (DavoudTaghawiNejad) [inline abcEconomics/trade.py]: Why net keeping slots for pypy/non cython compilation?

Comment (rht) [reply abcEconomics/trade.py]: It might slow down the cython version.

**Comment by DavoudTaghawiNejad** [comment]
Issue 176 must be merged in first. Its a nasty floating point error.

**Comment by rht** [comment]
The other option is to use .pxd files instead.

**Comment by rht** [comment]
Superseded by #182.

**Pull request closed without merging**